test/parallel/test-repl.js

```diff
@@ -23,551 +23,757 @@
+      ''
+    ]
+  },
+  // Mitigate https://github.com/nodejs/node/issues/548
+  {
+    send: 'function name(){ return "node"; };name()',
+    expect: '\'node\''
+  },
+  {
+    send: 'function name(){ return "nodejs"; };name()',
+    expect: '\'nodejs\''
+  },
+  // Avoid emitting repl:line-number for SyntaxError
+  {
+    send: 'a = 3.5e',
+    expect: [
+      kSource,
+      kArrow,
+      '',
+      /^SyntaxError: /,
+      ''
+    ]
+  },
+  // Avoid emitting stack trace
+  {
+    send: 'a = 3.5e',
+    expect: [
+      kSource,
+      kArrow,
+      '',
+      /^SyntaxError: /,
+      ''
+    ]
+  },
+
+  // https://github.com/nodejs/node/issues/9850
+  {
+    send: 'function* foo() {}; foo().next();',
+    expect: '{ value: undefined, done: true }'
+  },
+
+  {
+    send: 'function *foo() {}; foo().next();',
+    expect: '{ value: undefined, done: true }'
+  },
+
+  {
+    send: 'function*foo() {}; foo().next();',
+    expect: '{ value: undefined, done: true }'
+  },
+
+  {
+    send: 'function * foo() {}; foo().next()',
+    expect: '{ value: undefined, done: true }'
+  },
+
+  // https://github.com/nodejs/node/issues/9300
+  {
+    send: 'function foo() {\nvar bar = 1 / 1; // "/"\n}',
+    expect: '... ... undefined'
+  },
+
+  {
+    send: '(function() {\nreturn /foo/ / /bar/;\n}())',
+    expect: '... ... NaN'
+  },
+
+  {
+    send: '(function() {\nif (false) {} /bar"/;\n}())',
+    expect: '... ... undefined'
+  },
+
+  // https://github.com/nodejs/node/issues/16483
+  {
+    send: 'new Proxy({x:42}, {get(){throw null}});',
+    expect: 'Proxy [ { x: 42 }, { get: [Function: get] } ]'
+  },
+  {
+    send: 'repl.writer.options.showProxy = false, new Proxy({x:42}, {});',
+    expect: '{ x: 42 }'
+  },
+
+  // Newline within template string maintains whitespace.
+  {
+    send: '`foo \n`',
+    expect: '... \'foo \\n\''
+  },
+  // Whitespace is not evaluated.
+  {
+    send: ' \t  \n',
+    expect: 'undefined'
+  },
+  // Do not parse `...[]` as a REPL keyword
+  {
+    send: '...[]\n',
+    expect: '... ... '
+  },
+  // bring back the repl to prompt
+  {
+    send: '.break',
+    expect: ''
+  }
+];
+
+const tcpTests = [
+  {
+    send: '',
+    expect: ''
+  },
+  {
+    send: 'invoke_me(333)',
+    expect: '\'invoked 333\''
+  },
+  {
+    send: 'a += 1',
+    expect: '12346'
+  },
+  {
+    send: `require(${JSON.stringify(moduleFilename)}).number`,
+    expect: '42'
+  }
+];
+
+(async function() {
+  {
+    const [ socket, replServer ] = await startUnixRepl();
 
-    socket.on('end', function() {
+    await runReplTests(socket, prompt_unix, unixTests);
+    await runReplTests(socket, prompt_unix, errorTests);
+    replServer.replMode = repl.REPL_MODE_STRICT;
+    await runReplTests(socket, prompt_unix, strictModeTests);
+
+    socket.end();
+  }
+  {
+    const [ socket ] = await startTCPRepl();
+
+    await runReplTests(socket, prompt_tcp, tcpTests);
+
+    socket.end();
+  }
+})();
+
+function startTCPRepl() {
```

Comment:
@addaleax not sure why this didn't go through the first time, hopefully works this time!

Was just suggesting that maybe a factory method could be pulled out of `startTCPRepl` and `startUnixRepl`, it seems like there is some duplicated code there.

Reply:
@maclover7 Yeah, I agree :) It’s not 100 % trivial but it would sure be nice to have that! I might do it when I have the time but I think this is okay for now :)